File: hermes/src/main/java/org/eclipse/paho/android/service/MqttServiceConstants.java
/*******************************************************************************
 * Copyright (c) 1999, 2016 IBM Corp.
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * and Eclipse Distribution License v1.0 which accompany this distribution. 
 *
 * The Eclipse Public License is available at 
 *    http://www.eclipse.org/legal/epl-v10.html
 * and the Eclipse Distribution License is available at 
 *   http://www.eclipse.org/org/documents/edl-v10.php.
 */
package org.eclipse.paho.android.service;

/**
 * Various strings used to identify operations or data in the Android MQTT
 * service, mainly used in Intents passed between Activities and the Service.
 */
public interface MqttServiceConstants {

  /** 数据库字段-是否重复 */
  String DB_COLUMN_DUPLICATE = "duplicate";

  /** 数据库字段-是否服务器保留 */
  String DB_COLUMN_RETAINED = "retained";
  /** 数据库字段-消息的服务质量 */
  String DB_COLUMN_QOS = "qos";
  /** 数据库字段-数据内容 */
  String DB_COLUMN_PAYLOAD = "payload";
  /** 数据库字段-主题名称 */
  String DB_COLUMN_DESTINATION_NAME = "destinationName";
  /** 数据库字段-客户端标识符 */
  String DB_COLUMN_CLIENT_HANDLE = "clientHandle";
  /** 数据库字段-消息ID */
  String DB_COLUMN_MESSAGE_ID = "messageId";
  /** 数据库字段-消息到达时间 */
  String DB_COLUMN_MTIMESTAMP = "mtimestamp";
  /** 数据库消息表表名 */
  String DB_ARRIVED_MESSAGE_TABLE_NAME = "MqttArrivedMessageTable";




  //Intent prefix for Ping sender.
  String PING_SENDER = MqttService.TAG + ".pingSender.";
  
  //Constant for wakelock
  String PING_WAKELOCK = "MqttService.client.";

  // the name of the table in the database to which we will save messages

  String FILE_PERSISTENCE_DIR_NAME = "MqttConnection";

}

File: hermes/src/main/java/com/lodz/android/hermes/paho/android/service/MqttServiceConstants.kt
//package com.lodz.android.hermes.paho.android.service
//
//
///**
// * const val ious strings used to identify operations or data in the Android MQTT
// * service, mainly used in Intents passed between Activities and the Service.
// */
//object MqttServiceConstants {
//    /* Version information */
//    const val VERSION = "v0"
//
//    /* Attributes of messages <p> Used for the column names in the database */
//    const val DUPLICATE = "duplicate"
//    const val RETAINED = "retained"
//    const val QOS = "qos"
//    const val PAYLOAD = "payload"
//    const val DESTINATION_NAME = "destinationName"
//    const val CLIENT_HANDLE = "clientHandle"
//    const val MESSAGE_ID = "messageId"
//
//    /* Tags for actions passed between the Activity and the Service */
//    const val SEND_ACTION = "send"
//    const val UNSUBSCRIBE_ACTION = "unsubscribe"
//    const val SUBSCRIBE_ACTION = "subscribe"
//    const val DISCONNECT_ACTION = "disconnect"
//    const val CONNECT_ACTION = "connect"
//    const val CONNECT_EXTENDED_ACTION = "connectExtended"
//    const val MESSAGE_ARRIVED_ACTION = "messageArrived"
//    const val MESSAGE_DELIVERED_ACTION = "messageDelivered"
//    const val ON_CONNECTION_LOST_ACTION = "onConnectionLost"
//    const val TRACE_ACTION = "trace"
//
//    /* Identifies an Intent which calls back to the Activity */
//    const val CALLBACK_TO_ACTIVITY = MqttService.TAG + ".callbackToActivity" + "." + VERSION
//
//    /* Identifiers for extra data on Intents broadcast to the Activity */
//    const val CALLBACK_ACTION = MqttService.TAG + ".callbackAction"
//    const val CALLBACK_STATUS = MqttService.TAG + ".callbackStatus"
//    const val CALLBACK_CLIENT_HANDLE = MqttService.TAG + "." + CLIENT_HANDLE
//    const val CALLBACK_ERROR_MESSAGE = MqttService.TAG + ".errorMessage"
//    const val CALLBACK_EXCEPTION_STACK = MqttService.TAG + ".exceptionStack"
//    const val CALLBACK_INVOCATION_CONTEXT = MqttService.TAG + "." + "invocationContext"
//    const val CALLBACK_ACTIVITY_TOKEN = MqttService.TAG + "." + "activityToken"
//    const val CALLBACK_DESTINATION_NAME = MqttService.TAG + '.' + DESTINATION_NAME
//    const val CALLBACK_MESSAGE_ID = MqttService.TAG + '.' + MESSAGE_ID
//    const val CALLBACK_RECONNECT = MqttService.TAG + ".reconnect"
//    const val CALLBACK_SERVER_URI = MqttService.TAG + ".serverURI"
//    const val CALLBACK_MESSAGE_PARCEL = MqttService.TAG + ".PARCEL"
//    const val CALLBACK_TRACE_SEVERITY = MqttService.TAG + ".traceSeverity"
//    const val CALLBACK_TRACE_TAG = MqttService.TAG + ".traceTag"
//    const val CALLBACK_TRACE_ID = MqttService.TAG + ".traceId"
//    const val CALLBACK_ERROR_NUMBER = MqttService.TAG + ".ERROR_NUMBER"
//
//    const val CALLBACK_EXCEPTION = MqttService.TAG + ".exception"
//
//    //Intent prefix for Ping sender.
//    const val PING_SENDER = MqttService.TAG + ".pingSender."
//
//    //Constant for wakelock
//    const val PING_WAKELOCK = MqttService.TAG + ".client."
//    const val WAKELOCK_NETWORK_INTENT = MqttService.TAG + ""
//
//    //Trace severity levels
//    const val TRACE_ERROR = "error"
//    const val TRACE_DEBUG = "debug"
//    const val TRACE_EXCEPTION = "exception"
//
//    //exception code for non MqttExceptions
//    const val NON_MQTT_EXCEPTION = -1
//}